bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/type/TypeResolver.java

```diff
@@ -88,6 +91,34 @@ public static ThingType resolve(ThingTypeUID thingTypeUID, Locale locale) {
         return thingTypeRegistry != null ? thingTypeRegistry.getThingType(thingTypeUID, locale) : null;
     }
 
+    /**
+     * Resolves a {@link ConfigDescription} for the given {@link URI}.
+     *
+     * @param configDescriptionURI the URI of the configuration description
+     * @return config description or null if no config description was found or config description registry is not
+     *         present
+     */
+    public static ConfigDescription resolve(URI configDescriptionURI) {
```

Comment:
This does not really fit in the TypeResolver, which is rather for stuff about Things.
As this seems to be only used by the ThingManager and the ThingManager is an osgi service itself, you could simply add the ConfigDescriptionRegistry as an optional service there, couldn't you?